src/main/java/com/codejoust/main/service/GameManagementService.java

```diff
@@ -283,14 +307,131 @@ public GameDto manuallyEndGame(String roomId, EndGameRequest request) {
         return gameDto;
     }
 
-    protected void handleEndGame(Game game) {
+    public void handleEndGame(Game game) {
         // Cancel all previously scheduled timers
         GameTimer gameTimer = game.getGameTimer();
         gameTimer.getTimer().cancel();
 
         for (Timer timer : gameTimer.getNotificationTimers()) {
             timer.cancel();
         }
+
+        CreateGameReportTask createGameReportTask = new CreateGameReportTask(this, game);
+        Timer createGameReportTimer = new Timer();
+        createGameReportTimer.schedule(createGameReportTask, GameTimer.DURATION_1 * 1000);
```

Comment:
I'm sort of conflicted on this - I feel like timers always add extra complexity which could be a hassle down the line and also if teachers ever wanted to view their reports immediately after, it might be annoying to wait 1+ minutes or confusing if we don't have a banner informing them. 

I'm wondering if it makes sense to cut off submissions that go through after the deadline, that way we can immediately generate this in `handleEndGame`? The main downside I think is that students who submit with only a few seconds left might lose that submission, but imo that's an edge case and a risk they knew they were taking (and currently we also don't have a good way to show results that came in post-game over other than just recalculating the leaderboard, which might be confusing). Thoughts?

Reply:
I'm cool with that actually, just removing the `CreateGameReportTask` class and tests entirely and making that change mentioned. I also agree because I feel it'd be great to directly add a link to the game report - this won't happen instantaneously, but it could at least be calculated in a few seconds and it'd be easy to mock a "loading" symbol until that time. (In this case, I may also add a socket update to send once the game report is created, so we can display that update.)

I'll plan on doing that by only adding the submission if the game is not over - that way, the submission may be calculated, but if it doesn't complete before the game finishes, then it is not added and the game report would not change.

I will note, there's also the option in the future of letting the teacher choose. We may not want to bombard them with too many options, but if it becomes a bit issue that submissions are getting cut off from the game end, then this functionality could vary either way based on a room setting. For now though, I'm ok with this new implementation.

Reply:
I also added a boolean field to the `Game` object titled `createGameReportStarted` - this is true once the process of creating a game report for that specific game has begun. This means that if multiple of the end game conditions occurred at very close times, only one of the game reports should be saved on the database.

This is more reliable than checking if a game report with the game ID already exists in the database as this saving process could take a while, and this lag may end up accidentally saving two games. By doing the check locally, one of the end game conditions should set this field to true, preventing the other one from also creating a game report as that field's value is updated very quickly.